lib/src/main/java/com/futurewei/alcor/common/db/ignite/IgniteDbCache.java

```diff
@@ -142,6 +143,61 @@ public boolean remove(K key) throws CacheException {
         }
     }
 
+    @Override
+    public V get(Map<String, Object[]> filterParams) throws CacheException {
+        CachePredicate<String, BinaryObject> predicate = MapPredicate.getInstance(filterParams);
+        return get(predicate);
+    }
+
+    @Override
+    public <E1, E2> V get(CachePredicate<E1, E2> cachePredicate) throws CacheException {
+        QueryCursor<Cache.Entry<E1, E2>> cursor =
+                cache.withKeepBinary().query(ScanQueryBuilder.newScanQuery(cachePredicate));
+        List<Cache.Entry<E1, E2>> result = cursor.getAll();
+        if(result.size() > 1){
+            throw new CacheException("more than one rows found!");
+        }
+
+        if(result.isEmpty()){
+            return null;
+        }
+
+        E2 obj = result.get(0).getValue();
```

Comment:
How to ensure that there is only one result of the query? if so, is the getAll (CachePredicate < E1, E2 > cachePredicate) interface sufficient?